src/garage/envs/multi_env_wrapper.py

```diff
@@ -157,6 +157,7 @@ def reset(self, **kwargs):
         self.env = self._task_envs[self._active_task_index]
         obs = self.env.reset(**kwargs)
         oh_obs = self._obs_with_one_hot(obs)
+        self.env.spec.observation_space = self.observation_space
```

Comment:
this should be unchanging because all environments have the same observation and action spaces. did you find an example where this is not the case?

Reply:
it looks like you should instead makes the checks on L66 assert that everything is totally equal, rather than just having the same shape.